javascript/ql/src/semmle/javascript/frameworks/NodeJSLib.qll

```diff
@@ -150,6 +150,23 @@ module NodeJSLib {
     BuiltinRouteHandlerRequestExpr() { src instanceof RequestSource }
   }
 
+  /**
+   * A parameter containing data received by a NodeJS HTTP server.
+   * E.g. `chunk` in: `http.createServer().on('request', (req, res) => req.on("data", (chunk) => ...))`.
+   */
+  private class ServerRequestDataEvent extends RemoteFlowSource, DataFlow::ParameterNode {
+    RequestSource req;
```

Comment:
I'd suggest doing this with `HTTP::RequestSource` rather than the NodeJS-specific one. Some web frameworks, like Express, exposes request objects that are actually Node.js request objects with a modified prototype.